### ISSUE TITLE
Implement Gensim doc2vec

### ISSUE THREAD

**Issue by vkrgowtham** [RasaHQ/rasa_nlu#506]
Hi,
In both spacy model and mitie model, to represent a sentence only an average of word embeddings was implemented in the code. I know this gives decent accuracy, but how can we improve this ? Can we define some rule-based approximations to achieve better accuracy ? 
Do you have any plan to add more complex structures to represent sentence in future ? Why not use genism doc2vec which are better?
Thank you.

**Comment by PHLF** [comment]
> Why not use genism doc2vec which are better?

You know how overabused the word "better" is in the (software) engineering world 😄 

Thought vectors or doc2vec may perform "better" than mean average of word embeddings (we need figures!) but it will also increase the complexity cost to parse a sentence (adding another ML layer) thus reducing the number of parse requests a http server could handle (especially a Python one). 
So even if it could improve (a lot?) intent classification, I don't think it is a priority for the dev team. I'm not speaking for them so I may be completely mistaken.
Intent classification can also be improved a lot by increasing your data quality (simple utterances, specific and well identified vocab for each domain/intent...) at no performance cost.

But as you are now acquainted with the source code, feel free to try it for yourself and make a PR. I'm sure the Rasa dev team will be glad to have another great contribution 😉

**Comment by wrathagom** [comment]
I agree with @PHLF given the broadness of what the Rasa team has to work on at the moment seems like we need to quantify better before asking them to do it. That being said I'll add the help wanted label to this one and if someone has time to implement it as an alternative or configurable pipeline option, then great!

**Comment by amn41** [comment]
Unsupervised sentence representations like skip-thought, doc2vec, etc, are an interesting avenue, but as already guessed, not a priority right now. Classifying intents is something that works quite reliably, there are other areas where we'd like to improve performance more urgently. 

Also, we already provide *a lot* of options with the current set of components, and adding more options always comes with a cost for new users who don't know what to pick.

**Comment by vkrgowtham** [comment]
Hi,
I'm having problems with  classification of intents using (mitie+sklearn) and mitie pipelines. For example consider my training set consists of "search nearby shop", and user inputs some synonym for search like "locate shop" , the classifier is unable to recognise them. There are also instances which are getting wrongly classified like "I would like to search for nearest shop". I want know is there any method by which i can elevate preference to some words and suppress others(This should have already done by the feature vector). I have around 20 examples for each intent. Hence i was looking for different approaches to represent words or sentences.

Thank you.

**Comment by wrathagom** [comment]
If you'd be willing to provide your training data in a gist or e-mail we'd be happy to make suggestions on how to improve it.

No there is currently not a way to boost the importance of specific words. Have you tried a non-mitre based intent classifier?

**Comment by tmbo** [comment]
To boost the importance of certain words, you can use the pattern features, e.g. lets assume `shop` is an important word, an added regex feature can improve the intent classification performance (as part of the training data):
```json
{
  "rasa_nlu_data": {
    "regex_features": [
      {
        "name": "important_shop_word",
        "pattern": "shop"
      }
    ],
    ...
  }
}
```

**Comment by akelad** [comment]
i'm gonna close this, as this is not something we're going to implement
